types/signale.d.ts

```diff
@@ -0,0 +1,134 @@
+    displayLabel?: boolean;
+    /** Display the current local time in `HH:MM:SS` format. */
+    displayTimestamp?: boolean;
+    /** Underline the logger label. */
+    underlineLabel?: boolean;
+    /** Underline the logger message. */
+    underlineMessage?: boolean;
+  }
+
+  interface SignaleOptions<TTypes extends string = DefaultMethods> {
+    /** Sets the configuration of an instance overriding any existing global or local configuration. */
+    config?: SignaleConfig;
+    /**
+     * Name of the scope.
+     */
+    scope?: string;
+    /**
+     * Holds the configuration of the custom and default loggers.
+     */
+    types?: Partial<Record<TTypes, CommandType>>;
+    interactive?: boolean;
+    timers?: Map<string, Date>;
+    /**
+     * Destination to which the data is written, can be any valid
+     * [Writable stream](https://nodejs.org/api/stream.html#stream_writable_streams).
+     */
+    stream?: NodeJS.WriteStream;
+  }
+
+  interface SignaleConstructor {
+    new <TTypes extends string = DefaultMethods>(
+      options?: SignaleOptions<TTypes>
+    ): Signale<TTypes>;
+  }
+
+  interface SignaleBase<TTypes extends string = DefaultMethods> {
+    /**
+     * Sets the configuration of an instance overriding any existing global or local configuration.
+     *
+     * @param configObj Can hold any of the documented options.
+     */
+    config(configObj: SignaleConfig): Signale<TTypes>;
+    /**
+     * Defines the scope name of the logger.
+     *
+     * @param name Can be one or more comma delimited strings.
+     */
+    scope(...name: string[]): Signale<TTypes>;
+    /** Clears the scope name of the logger. */
+    unscope(): void;
+    /**
+     * Sets a timers and accepts an optional label. If none provided the timer will receive a unique label automatically.
+     *
+     * @param label Label corresponding to the timer. Each timer must have its own unique label.
+     * @returns a string corresponding to the timer label.
+     */
+    time(label?: string): string;
+    /**
+     * Deactivates the timer to which the given label corresponds. If no label
+     * is provided the most recent timer, that was created without providing a
+     * label, will be deactivated.
+     *
+     * @param label Label corresponding to the timer, each timer has its own unique label.
+     * @param span Total running time.
+     */
+    timeEnd(label?: string, span?: number): { label: string; span?: number };
+  }
+
+  type LoggerFunc = (message?: any, ...optionalArgs: any[]) => void;
+  type Signale<TTypes extends string = DefaultMethods> = SignaleBase<TTypes> &
+    Record<TTypes, LoggerFunc> &
+    Record<DefaultMethods, LoggerFunc>;
+}
+
+declare const signale: signale.Signale<signale.DefaultMethods> & {
+  Signale: signale.SignaleConstructor;
+  SignaleConfig: signale.SignaleConfig;
+  SignaleOptions: signale.SignaleOptions;
+  DefaultMethods: signale.DefaultMethods;
+};
+
+export = signale;
+export as namespace signale;
```

Comment:
The `export as namespace` declaration is used with modules that export some sort of global variable. See [here](https://stackoverflow.com/a/44849515/1332403) and [here](https://github.com/zspitz/TypeScript-Handbook/blob/release-2.0/pages/Modules.md#umd-modules). Since Signale appears to be a CommonJS-only module and doesn't export a global, this line can probably be removed.

Reply:
Fixed. 👌